.changeset/dull-lamps-invent.md

```diff
@@ -0,0 +1,17 @@
+---
+"@apollo/query-planner": patch
+"@apollo/query-graphs": patch
+"@apollo/composition": patch
+"@apollo/federation-internals": patch
+"@apollo/gateway": patch
```

Comment:
Thinking this should be `minor` since we're treating this as a new feature and landing to `next`

```suggestion
"@apollo/query-planner": minor
"@apollo/query-graphs": minor
"@apollo/composition": minor
"@apollo/federation-internals": minor
"@apollo/gateway": minor
```